django/researchdata/models.py

```diff
@@ -0,0 +1,332 @@
+                                        null=True,
+                                        verbose_name='holding location (search)')
+    holding_location_coordinates = PlainLocationField(based_fields=['holding_location'],
+                                                      zoom=7,
+                                                      blank=True,
+                                                      null=True,
+                                                      verbose_name='holding location (coordinates)')
+
+    # Admin and meta fields
+    admin_notes = models.TextField(blank=True, null=True)
+    meta_created_datetime = models.DateTimeField(auto_now_add=True, verbose_name='Created')
+    meta_lastupdated_datetime = models.DateTimeField(auto_now=True, verbose_name='Last Updated')
+
+    def __str__(self):
+        return self.name
+
+
+class Person(models.Model):
+    """
+    A person that's associated with other data, e.g. an author of an Item
+    """
+
+    title = models.ForeignKey(SlPersonTitle, on_delete=models.SET_NULL, blank=True, null=True)
+    first_name = models.CharField(max_length=255, blank=True, null=True)
+    last_name = models.CharField(max_length=255, blank=True, null=True)
+    other_names = models.CharField(max_length=255, blank=True, null=True)
+    birth_date = models.DateField(blank=True, null=True)
+    birth_location = models.CharField(max_length=255, verbose_name='birth location (search)')
+    birth_location_coordinates = PlainLocationField(based_fields=['birth_location'],
+                                                    zoom=7,
+                                                    blank=True,
+                                                    null=True,
+                                                    verbose_name='birth location (coordinates)')
+    death_date = models.DateField(blank=True, null=True)
+    death_location = models.CharField(max_length=255, verbose_name='death location (search)')
+    death_location_coordinates = PlainLocationField(based_fields=['death_location'],
+                                                    zoom=7,
+                                                    blank=True,
+                                                    null=True,
+                                                    verbose_name='death location (coordinates)')
+
+    # Admin and meta fields
+    admin_notes = models.TextField(blank=True, null=True)
+    meta_created_datetime = models.DateTimeField(auto_now_add=True, verbose_name='Created')
+    meta_lastupdated_datetime = models.DateTimeField(auto_now=True, verbose_name='Last Updated')
+
+    @property
+    def name(self):
+        names = [name for name in [self.title, self.first_name, self.last_name] if name]
+        return " ".join(names) if len(names) else f"(no name, ID: {self.id})"
+
+    def __str__(self):
+        return self.name
+
+
+# Relationship tables (M2M with additional fields)
+
+
+class RelEntityAndEvent(models.Model):
```

Comment:
The 'Rel...' naming looks like a good idea for relationships. Is this a Django convention (or a Mike convention)? Either way, I think I might steal it.

Reply:
Rel is just a Mike convention. Same as Sl for 'select list' models. No a Django convention, just a personal one.